flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -482,6 +511,67 @@ void reassignPartitions(List<KafkaTopicPartitionState<TopicPartition>> newPartit
 		return new KafkaConsumer<>(kafkaProperties);
 	}
 
+	@VisibleForTesting
+	RateLimiter getRateLimiter() {
+		return rateLimiter;
+	}
+
+	// -----------------------------------------------------------------------
+	// Rate limiting methods
+	// -----------------------------------------------------------------------
+
+	/**
+	 * Set the {{@link #bytesPerSecondMax}} per consumer using the global bytes per second.
+	 */
+	public void setBytesPerSecondMax() {
+		// TODO: Pass runtime context to get bytesPerSecondMax for this subtask.
+		long globalBytesPerSecondMax = Long.valueOf(kafkaProperties
+				.getProperty(BYTES_PER_SECOND_MAX, Long.toString(DEFAULT_BYTES_PER_SECOND_MAX)));
+		this.bytesPerSecondMax = globalBytesPerSecondMax;
+	}
+
+	/** Create RateLimiter with a rate of {{@link #bytesPerSecondMax}}.
+	 *
+	 */
+	private void createRateLimiter() {
```

Comment:
Suggestion: Call this getRateLimiter() or getOrCreateRateLimiter() just to help with readability.

Reply:
Updated to getOrCreateRateLimiter().